### PR TITLE
Binance websocket orderbook sync fix

### PR DESCRIPTION
**Pull request by vazha**
# PR Description

As I wrote in Slack - PR #599  does not fixed issues with Binance websocket orderbook syncronisattion.
I came to the conclusion that the reason is that the subscription to the order book by websockets occurs after the getting snapshot of the orderbook received by the REST API. I mean The LastUpdateID (of smapshot) is less than the FirstUpdateID of the websocket updates. Thus, some of the updates are lost, which leads to freezing of prices in the order book.
I have moved part of code to different goroutine, to let websocket subscripton starts earlyer. I have tested this solution many times and it works fine.

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

**Review by gloriousCode — APPROVED** [review]
Thanks for this PR @vazha, This is an improvement given that we could only subscribe to websocket orderbook updates once all REST requests were done, which could be awhile one has many currencies enabled. I went back and tested and was able to get out of sync orderbooks and I have not since using your fix.

File: exchanges/binance/binance_websocket.go [hunk 1]
```diff
@@ -84,21 +84,28 @@ func (b *Binance) WsConnect() error {
...
-		return err
+		log.Errorf(log.ExchangeSys, "orderBookProcess, GetEnabledPairs error: %s", err)
+		return
 	}
 
 	for i := range enabledPairs {
 		err = b.SeedLocalCache(enabledPairs[i])
 		if err != nil {
-			return err
+			log.Errorf(log.ExchangeSys, "orderBookProcess, SeedLocalCache error: %s", err)
```
Comment (thrasher-) [inline exchanges/binance/binance_websocket.go:102]:
```suggestion
		log.Errorf(log.ExchangeSys, "%s orderBookProcess, SeedLocalCache error: %s", b.Name, err)
```

**Review by thrasher- — CHANGES_REQUESTED** [review]
Thanks for this PR! Only have a basic nit due to the ambiguity of having no exchange name associated with the error message when multiple exchanges are enabled

**Review by gloriousCode — APPROVED** [review]
Thanks for making those changes!

**Review by MadCozBadd — APPROVED** [review]
tACK!!!!!

File: exchanges/binance/binance_websocket.go [hunk 2]
```diff
@@ -84,21 +84,28 @@ func (b *Binance) WsConnect() error {
...
+	go b.orderBookProcess()
+	return nil
+}
+
+// orderBookProcess prepare orderbook handling
+func (b *Binance) orderBookProcess() {
 	enabledPairs, err := b.GetEnabledPairs(asset.Spot)
 	if err != nil {
-		return err
+		log.Errorf(log.ExchangeSys, "orderBookProcess, GetEnabledPairs error: %s", err)
```
Comment (thrasher-) [inline exchanges/binance/binance_websocket.go:95]: Just missing one more b.Name here and then this is good to merge!

**Review by thrasher- — CHANGES_REQUESTED** [review]

**Review by MadCozBadd — APPROVED** [review]

**Review by gloriousCode — APPROVED** [review]

**Review by thrasher- — APPROVED** [review]
Thanks @vazha !

**Pull request merged**